cmake/finds/RequiredDependencySearch.cmake

```diff
@@ -79,6 +78,17 @@ if(MUQ_USE_OPENMPI)
 
 endif()
 
+############################
+##### LOOK FOR NLOPT  ######
+############################
+
+set(HAVE_NLopt 1)
+
+find_package(NLopt REQUIRED)
+if (NLopt_FOUND)
+    set(MUQ_HAS_NLopt 1)
+endif ()
```

Comment:
i don't understand this: if this is required how can the if afterward be any useful?

Reply:
please address as discussed